### PR TITLE
feat!: switch template from nunjucks to react rendering engine

### PR DESCRIPTION
**Pull request by kaushik-rishi**
**Description**

- Switch the template from nunjucks to react rendering engine

**Things to verify before merging**
- [x] verify if scripts in `package.json` are running fine
- [x] update any/all **snapshot tests**
- [x] test the generated template for mqtt and mqtts
- [x] test kafka template snapshot tests
- [x] test by providing multiple channels for a single subscribe / publish
- [ ] decide on new version to bump into (breaking change)

**Related issue(s)**
Issue #133

### CODE REVIEW

File: utils/channels-topics.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import { URL } from "url";
```
Comment (derberg) [inline utils/channels-topics.js:1]: lets follow best practice from https://github.com/asyncapi/template-for-generator-templates and have `helpers` and not `utils`

File: template/src/api/routes/$$channel$$.js [hunk 2]
```diff
@@ -1,79 +1,126 @@
-const Router = require('hermesjs/lib/router');
-const {validateMessage} = require('../../lib/message-validator');
-const router = new Router();
-const {{ channelName | camelCase }}Handler = require('../handlers/{{ channelName | convertToFilename }}');
-module.exports = router;
+import { File } from "@asyncapi/generator-react-sdk";
```
Comment (derberg) [inline template/src/api/routes/$$channel$$.js:1]: why with react you still want to use `file templates` (`$$channel$$.js`) instead new way? look at https://github.com/asyncapi/generator/pull/1117#issuecomment-2006369580 for more details

**Review by derberg — COMMENTED** [review]
please also make sure `__transpiled` is in `.gitignore`

**Review by derberg — APPROVED** [review]
great work @kaushik-rishi 

we need to do breaking change as https://github.com/asyncapi/nodejs-template/pull/267/commits/3658766909644963531a8c496475f4bfa6deee2d had to be done, to make sure generator with react engine is required. I also made sure to require minimum 1.15 that is the first version enabling v3 that you anyway need

2 followup changes are needed, in separate PR:
- `template/README.md.js` has outdated instructions
- https://github.com/asyncapi/nodejs-template/?tab=readme-ov-file#how-to-use-the-template is also outdated, `npm start` no longer makes sense. So combine some of the steps (but use AsyncAPI CLI) with https://github.com/asyncapi/nodejs-template/?tab=readme-ov-file#adding-custom-code--handlers that are fine - so it is end to end copy paste instruction
- maybe it makes sense to have script from `Adding custom code / handlers` section actually stored in `test` and have a script that starts it, like `npm run example`

**Comment by derberg** [comment]
@kaushik-rishi sorry, had to update generator in dev dep

**Comment by kaushik-rishi** [comment]
> great work @kaushik-rishi 
> 
> 
> 
> we need to do breaking change as https://github.com/asyncapi/nodejs-template/pull/267/commits/3658766909644963531a8c496475f4bfa6deee2d had to be done, to make sure generator with react engine is required. I also made sure to require minimum 1.15 that is the first version enabling v3 that you anyway need
> 
> 
> 
> 2 followup changes are needed, in separate PR:
> 
> - `template/README.md.js` has outdated instructions
> 
> - https://github.com/asyncapi/nodejs-template/?tab=readme-ov-file#how-to-use-the-template is also outdated, `npm start` no longer makes sense. So combine some of the steps (but use AsyncAPI CLI) with https://github.com/asyncapi/nodejs-template/?tab=readme-ov-file#adding-custom-code--handlers that are fine - so it is end to end copy paste instruction
> 
> - maybe it makes sense to have script from `Adding custom code / handlers` section actually stored in `test` and have a script that starts it, like `npm run example`

Thanks for the review and generator version update.
Will do the required changes and push.

**Review by derberg — APPROVED** [review]
@kaushik-rishi please do readme changes in separate

**Pull request merged**